repos/fdbt-site/src/interfaces/typeGuards.ts

```diff
@@ -178,3 +179,11 @@ export const isMultiOperatorMultipleServicesTicket = (
     (ticket as MultiOperatorMultipleServicesTicket).additionalOperators &&
     (ticket as MultiOperatorMultipleServicesTicket).additionalOperators.length > 0 &&
     'selectedServices' in ticket;
+
+export const isFlatFarePricedByDistanceTicket = (
+    ticket: Ticket | WithIds<Ticket>,
+): ticket is FlatFareMultipleServices =>
```

Comment:
the function is called isFlatFarePricedByDistanceTicket but the code inside it doesnt confirm if its priced by distance. The function should probably be `isPricedByDistanceProduct` and sets the product as PriceByDistanceProductDetails.